src/dune_pkg/mount.ml

```diff
@@ -50,7 +50,7 @@ let of_opam_url loc url =
          Path.relative dir file_digest
        in
        let+ path =
-         Tar.extract ~archive ~target
+         Archive_driver.extract Archive_driver.tar ~archive ~target
```

Comment:
Here, I think you need to call `choose_for_filename` too. Indeed, if you point to a `zip` archive, you will not be able to extract the `opam` file from it.

Reply:
Is it correct to say that this will only be an issue if an opam repository is stored in a .zip archive? It won't have any effect on package sources in zip files, right? I'll change it anyway, but just want to make sure I understand the implication of this.

Reply:
I wasn't thinking about opam repository, more about when you pin files in that are stored in a zip. In the current (inefficient but functional) version I implemented, it will download the zip file to get the `*.opam` file before generating the correct `.pkg` file associated. If you didn't add the check for the file type, it would prevent pinning projects stored in a `zip`.

Thus, the version that has been merged is correct :+1: